jwst/resample/resample.py

```diff
@@ -54,6 +58,14 @@ def __init__(self, input_models, output=None, **pars):
         # Define output WCS based on all inputs, including a reference WCS
         self.output_wcs = resample_utils.make_output_wcs(self.input_models)
         log.debug('Output mosaic size: {}'.format(self.output_wcs.data_size))
+        can_allocate, required_memory = datamodels.util.check_memory_allocation(
+            self.output_wcs.data_size, pars['allowed_memory'], datamodels.ImageModel
+        )
+        if not can_allocate:
+            raise RuntimeError(
```

Comment:
I'd recommend a dedicated exception class for this, so we're not converting all `RuntimeError` to SKIP

Reply:
Though a comment on this: the stpipe architecture has some sort of global try...except thing. I will see what that is really supposed to be doing. Otherwise, agreed.